kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtl.scala

```diff
@@ -26,3 +35,88 @@ private[ctl] object KyuubiCtlActionService extends Enumeration {
   type KyuubiCtlActionService = Value
   val SERVER, ENGINE = Value
 }
+
+/**
+ * Main gateway of lauching a Kyuubi Ctl action.
+ * See usage in [[KyuubiCtlArguments.printUsageAndExit]].
+ */
+private[kyuubi] class KyuubiCtl extends Logging {
+  import ServiceDiscovery._
+
+  def doAction(args: Array[String]): Unit = {
+    val ctlArgs = parseArguments(args)
+    if (ctlArgs.verbose) {
+      info(ctlArgs.toString)
+    }
+    ctlArgs.action match {
+      case KyuubiCtlAction.CREATE => create(ctlArgs)
+      case KyuubiCtlAction.HELP => printUsage(ctlArgs)
+      // TODO: support other actions
+      case _ =>
+    }
+  }
+
+  protected def parseArguments(args: Array[String]): KyuubiCtlArguments = {
+    new KyuubiCtlArguments(args)
+  }
+
+  /**
+   * Create a zookeeper service node.
+   */
+  private def create(args: KyuubiCtlArguments): Unit = {
+    val zkClient = getZkClient(args)
+    val instance = s"${args.host}:${args.port}"
+    val namespace = args.service match {
+      case KyuubiCtlActionService.ENGINE =>
+        // the logic is replicated from SQLEngineAppName.getZkNamespace for USER shared level
+        ZKPaths.makePath(s"${args.nameSpace}_${ShareLevel.USER}", args.user)
+      case KyuubiCtlActionService.SERVER => args.nameSpace
+    }
+    createZkServiceNode(zkClient, namespace, instance, args.version)
+  }
+
+  private def printUsage(args: KyuubiCtlArguments): Unit = {
+    args.printUsageAndExit(0)
+  }
+
+  private def getZkClient(args: KyuubiCtlArguments): CuratorFramework = {
+    val conf = new KyuubiConf().loadFileDefaults()
```

Comment:
can we inline KyuubiConf in `KyuubiCtlArguments` then we don't need to load the file twice

Reply:
sure